executor/aggregate.go

```diff
@@ -1218,14 +1232,18 @@ type StreamAggExec struct {
 
 // Open implements the Executor Open interface.
 func (e *StreamAggExec) Open(ctx context.Context) error {
-	if err := e.baseExecutor.Open(ctx); err != nil {
-		return err
-	}
 	failpoint.Inject("mockStreamAggExecBaseExecutorOpenReturnedError", func(val failpoint.Value) {
 		if val.(bool) {
 			failpoint.Return(errors.New("mock StreamAggExec.baseExecutor.Open returned error"))
 		}
 	})
+
+	if err := e.baseExecutor.Open(ctx); err != nil {
+		return err
+	}
+	// If panic in Open, the children executor should be closed because they are open.
+	defer closeBaseExecutor(&e.baseExecutor)
```

Comment:
For the panic in `Open` case, we do not handle the resource release well.

Reply:
Actually here is not just a problem in the test code, it's also a bug of the product.

```
set @@tidb_memtracker_size = xxx;
insert into select * from t1 where a in select b from t2 ... 
// If Out of memory, panic in Open, then there will be goroutine leak 
```